specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,132 +1,224 @@
+                "description": "unique schema identifier"
+              },
+              "X-Schema-Id-Location": {
+                "type": "string",
+                "description": "location of schema resource"
+              },
+              "X-Schema-Version": {
+                "type": "integer",
+                "description": "version of returned schema"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/SchemaId"
+            }
+          }
+        }
+      },
+      "put": {
+        "summary": "Register new schema",
+        "description": "Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.\n",
+        "x-ms-examples": {
+          "Register schema": {
+            "$ref": "./examples/registerSchema.json"
+          }
+        },
+        "operationId": "registerSchema",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "group-name",
+            "in": "path",
+            "description": "schema group",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "schema-name",
+            "in": "path",
+            "description": "schema name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "schema-content",
+            "in": "body",
+            "description": "schema content",
+            "required": true,
             "schema": {
-              "$ref": "#/definitions/Result"
+              "type": "string"
             }
           },
-          "default": {
-            "description": "Error response describing why the operation failed.",
+          {
+            "name": "serialization-type",
+            "in": "header",
+            "description": "serialization type for schema being registered",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string"
+              },
+              "X-Serialization": {
+                "type": "string",
+                "description": "schema type"
+              },
+              "X-Schema-Id": {
+                "type": "string",
+                "description": "unique schema identifier"
+              },
+              "X-Schema-Id-Location": {
+                "type": "string",
+                "description": "location of schema resource"
+              },
+              "X-Schema-Version": {
+                "type": "integer",
+                "description": "version of returned schema"
+              }
+            },
             "schema": {
-              "$ref": "#/definitions/ErrorResponse"
```

Comment:
Is there no strongly-typed error response model for your APIs?

Reply:
Error is returned as XML from the service, so typing error response as a string.